src/runtime/virtcontainers/qemu_arch_base.go

```diff
@@ -43,6 +43,15 @@ type qemuArch interface {
 	// machine returns the machine type
 	machine() govmmQemu.Machine
 
+	// protection returns guest protection type
+	guestProtection() guestProtection
+
+	// set attestation id for SEV guests with pre-attestation
+	setSEVAttestationId(attestation_id string)
```

Comment:
Go argument name style: 

```suggestion
	setSEVAttestationId(attestationId string)
```

(and similarly in the function implementation)